### PR TITLE
Maturing the extensions pattern

### PR DESCRIPTION
**Pull request by Ssukriti**
For issue : https://github.com/InnerSourceCommons/InnerSourcePatterns/issues/487#issuecomment-1325523147

This PR updates the initial pattern so we can move it to structured

### CODE REVIEW

**Comment by spier** [comment]
Great iteration @Ssukriti. I like the visual!

Will read this in more detail in the coming days and provide feedback then.

One question already:
As part of the description of IBM under "Known Instances", would it make sense to reference any other online material where IBM has been talking about this story?

For example Bill Higgins' talk at the last InnerSource Summit:
https://youtu.be/Lz-tIc2cyRM

**Comment by spier** [comment]
Hi @Ssukriti.

I have been sick for a couple of days, so didn't have time to review this yet. I will eventually though. Of course anybody else here is also free to review this in the meantime :)

Just two points that had come to mind about this pattern in general:

**visual**

How could we get your visual into a form that we can keep improving it, as the pattern matures? If we only include the `.png` then that is a bit hard to do. We haven't solved this problem for other patterns yet either. So this is mostly exploring what solutions other people know about. 

I have tried [one approach](https://github.com/InnerSourceCommons/InnerSourcePatterns/tree/main/assets/img/standard-base-documentation) recently which is basically: dedicated folder for the image incl a README that points to the original google drive document. That original is editable, if you have access. If you have seen any working approach for this problem elsewhere, I would love to learn about it.

**relation to Incubator Pipeline** 

Do you see any similarities between your pattern here and the [Incubator Pipeline](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/incubator-pipeline.md)? I have wondering if your plugin mechanism can been seen as some sort of "incubation period" that functionality has go through before it potentially gets the chance to make it into the core. 

_Note:_ The Incubator Pipeline is still a draft pattern, as it doesn't have known instances yet. However if it helps we can ask @gyehuda for his input, as he submitted that pattern.

**Comment by Ssukriti** [comment]
@spier on the 3 things mentioned above,

**1. Link to online material :** Sure I have added that talk from Bill Higgins .

 There are also similar architectures to extensions in open source : extensions to tensorflow https://www.tensorflow.org/resources/libraries-extensions , plugins to spacy: https://spacy.io/universe , do you think we can mention that here?

**2. Visual images:** I like your idea and can follow the same to allow the community to also edit the image . I have added that to PR

**3. Incubation pattern:** Yes extensions does provide a way for offering an incubation period before new contributions can make it to shared library. It addresses all concerns mentioned in incubation pattern. 

Quote from our pattern: 
`**Extensions provide a filter for new capabilities that may eventually move into the core of the project.** Extensions also act as an **incubation** and community hardening environment, allowing for much of that hardening to happen organically rather than in a costly review process.`

we have mentioned in extensions lifecycle :
    **Create criteria for when an extension can be ported to the primary repository, such as adoption of the extension by internal products and demand for the feature.
    Porting process of the extension to the primary repository will follow more stringent code review guidelines as set by library maintainers.**

By contribution new features external to common shared library using extensions:
1. We provide a way for reuse of common infrastructure and boilerplate code from main library, without adding more code and dependencies into the main library. **So contributors can just focus on adding the new features** and not rewrite entire infrastructure . It also provides a consistent user experience with the same APIs.
2. It also provides a community scaling mechanism since extension owners are responsible for providing support and maintaining extension.
3. It allows easy porting into main shared library once extension has gained some use cases, since infrastructure and APIs will be  the same and only code quality will need to be improved for the feature added. 

This has several advantages over creating completely new repositories from scratch to add new capabilities

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+An InnerSource project is receiving too many contributions, making maintenance difficult. By offering an extension mechanism outside of the core project, the maintainers enable scaling of project capabilities with minimal cost and maintenance overhead.
 
 ## Problem
 
-As the number of new feature contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
+As the number of contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. This results in a large code review backlog or premature rejection of new feature contributions. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
As the number of contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. This results in a large code review backlog or premature rejection of new feature contributions.

How can the host team allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
```

Replaced "we" with "host team", to make it more clear who is pushing for this improvement.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 2]
```diff
@@ -73,7 +91,7 @@ Following the principles outlined above ensures that:
 
 ## Known Instances
 
-**IBM Corporation** has adopted this solution to scale InnerSource AI libraries. Using extensions, developers are able to extend AI libraries with more algorithms and share their innovations with the community. The core libraries only contain strategic algorithms that have been adopted and validated, keeping them easier to maintain as we scale contributions.
+**IBM Corporation** has adopted this solution to scale [InnerSource AI libraries](https://youtu.be/Lz-tIc2cyRM). Using extensions, developers are able to extend AI libraries with more algorithms and share their innovations with the community. The core libraries only contain strategic algorithms that have been adopted and validated, keeping them easier to maintain as we scale contributions.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
* **IBM Corporation** has adopted this solution to scale [InnerSource AI libraries](https://youtu.be/Lz-tIc2cyRM). Using extensions, developers are able to extend AI libraries with more algorithms and share their innovations with the company-internal community. The core libraries only contain strategic algorithms that have been adopted and validated, keeping them easier to maintain as we scale contributions.

## Alias

Extensions to Manage Contributions at Scale
```

Let's make this a 1-bullet list, to highlight that we want to collect a list of known instances here.

Also added the Alias that I mentioned before.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 3]
```diff
@@ -1,23 +1,39 @@
 ## Title
 
-Extensions to Manage Contributions at Scale
+Extensions for Sustainable Growth
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md:3]: I like the new title!

We can also add the old title as an "Alias" at the bottom of the pattern.
That way the pattern can also be found via those keywords.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
+As the number of contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. This results in a large code review backlog or premature rejection of new feature contributions. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
+
+## Story
+
+There is a strategic project that aims to collect the best innovations within a domain space to one common stack, allowing reuse of a common infrastructure and providing a standard user experience. Through InnerSource, various teams in the organization that work within the domain space get an opportunity to collaborate and contribute their innovations to the common codebase. However, a large number of contributions in parallel from several developers is making maintenance of the codebase difficult. This is adding a huge burden on the project maintainers who assume ownership over the code quality standards and enable the community through various forms of communication.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
There is a strategic project that aims to collect the best innovations within a domain space to one common stack, allowing reuse of a common infrastructure and providing a standard user experience. Through InnerSource, various teams in the organization that work within the domain space get an opportunity to collaborate and contribute their innovations to the common codebase.

However, a large number of contributions in parallel from several developers is making maintenance of the codebase difficult. This is adding a huge burden on the project maintainers who assume ownership over the code quality standards and enable the community through various forms of communication.
```

Seems like a natural place to split this rather long paragraph.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 5]
```diff
@@ -1,23 +1,39 @@
...
+
+Project maintainers are burnt out due to:
+
+- Everlasting backlog of pull requests that need to be reviewed.
+- Job dissatisfaction: Majority of maintainers' time spent in community support leaves no room for innovation.
+- Perceived lack of accomplishment: Not all contributed features have adequate user demand and result in consequent adoption.
+- Time consuming releases: More features in the codebase results in long running tests.
+- Increase in maintenance activities: More bugs raised as new capabilities are added.
+
+A lot of time and investment is spent on maturing every new feature contribution, before potential users even get an opportunity to explore the features for their use cases.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
A lot of time is spent on maturing every new feature contribution, before potential users even get an opportunity to explore the feature for their use cases. If it turns out that new feature isn't fulfilling the use case, then all that time spent on achieving the desired code quality standards are waste.
```

I elaborated on your sentence here to make it more explicit what the waste is.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 6]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
...
 - The maintainer burden is reduced until an extension has demonstrated that it fills an important gap in the primary project.
 - The core project's common code (e.g. base classes and utility functions) can be a starting point for new development that extends project's domain. This avoids the need to port innovative work after-the-fact, thus reducing the overall burden of developing novel features for the project.
 - Developers are more likely to contribute and stay involved in maintenance and building communities for their codebase, which is also good for the health of the overall project ecosystem.
 
 ## Resulting Context
 
-- The project is able to scale with the addition of new features, without adding a maintenance overhead on the primary project repository.
+- The project is able to scale with the addition of new features, without adding maintenance overhead on the primary project repository.
 - Faster release of new features and experimental features for the community to explore, encouraging innovation and experimentation.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]: Do you mean "new features" via the core and "experimental features" via the extensions?

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 7]
```diff
@@ -1,23 +1,39 @@
...
+As the number of contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. This results in a large code review backlog or premature rejection of new feature contributions. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
+
+## Story
+
+There is a strategic project that aims to collect the best innovations within a domain space to one common stack, allowing reuse of a common infrastructure and providing a standard user experience. Through InnerSource, various teams in the organization that work within the domain space get an opportunity to collaborate and contribute their innovations to the common codebase. However, a large number of contributions in parallel from several developers is making maintenance of the codebase difficult. This is adding a huge burden on the project maintainers who assume ownership over the code quality standards and enable the community through various forms of communication.
+
+Project maintainers are burnt out due to:
+
+- Everlasting backlog of pull requests that need to be reviewed.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
- Everlasting backlog of pull requests from contributors that need to be reviewed.
```

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 8]
```diff
@@ -1,23 +1,39 @@
...
-- With the high volume of new capabilities being added, the organization is investing large amounts of time on code review cycles to harden the capabilities before release.  Not all of these capabilities gain adoption as they may not serve an internal use case.
-- Adding an excessive number of capabilities and code to the strategic repository is making it difficult to maintain.
-- As the maintainers cannot keep up with feedback to the contributors and code reviews anymore it creates a growing backlog of new features and ideas for the project which has scaled.
+- A strategic InnerSource codebase is scaling rapidly with new feature contributions from several employees.
+- The ratio of reviewers to contributions results in a growing backlog of pull requests. This is slowing down release of new features to the community.
+- Quality of the codebase is no longer maintained and user experience is adversely impacted.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
- Quality of the codebase is degrading and user experience is adversely impacted.
```

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 9]
```diff
@@ -28,7 +44,7 @@ As the number of new feature contributions to a mature InnerSource repository ra
 
 ## Solutions
 
-Allowing extensions/plugins to high-scale InnerSource codebases can relieve the maintenance burden on repository maintainers and allow faster release of new features for adopting products to explore. This shifts maintenance of capabilities to extension owners and allows the primary repository to support capabilities that have been adopted more widely and are more strategic.  
+Allowing [extensions/plugins](https://en.wikipedia.org/wiki/Extensibility) to high-scale InnerSource codebases can relieve the maintenance burden on repository maintainers and allow faster release of new features for adopting products to explore. This shifts maintenance of capabilities to extension owners and allows the primary repository to support capabilities that have been adopted more widely and are more strategic.  
 
 Extensions provide a filter for new capabilities that may eventually move into the core of the project. Extensions also act as an incubation and community hardening environment, allowing for much of that hardening to happen organically rather than in a costly review process.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md:53]: What do you mean by "filter"?

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 10]
```diff
@@ -73,7 +91,7 @@ Following the principles outlined above ensures that:
 
 ## Known Instances
 
-**IBM Corporation** has adopted this solution to scale InnerSource AI libraries. Using extensions, developers are able to extend AI libraries with more algorithms and share their innovations with the community. The core libraries only contain strategic algorithms that have been adopted and validated, keeping them easier to maintain as we scale contributions.
+**IBM Corporation** has adopted this solution to scale [InnerSource AI libraries](https://youtu.be/Lz-tIc2cyRM). Using extensions, developers are able to extend AI libraries with more algorithms and share their innovations with the community. The core libraries only contain strategic algorithms that have been adopted and validated, keeping them easier to maintain as we scale contributions.
 
 ## Status
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md:104]: You can already change the status to **Structured**.

Once all comments are resolved, you can then `git mv` the file to the `2-structured` folder as well.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 11]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
    - Create criteria for when an extension can be ported to the primary repository, such as adoption of the extension by internal products and demand for the feature.
    - Porting process of the extension to the primary repository will follow more stringent code review guidelines as set by library maintainers.
 
+![Project extensions](../../assets/img/extensions-for-sustainable-growth/extensions-for-sustainable-growth.png)
+
 Following the principles outlined above ensures that:
 
-- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of boilerplate code.
+- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) code.
 - Extensions are discoverable in a repeatable manner to all users of the primary project; just because code doesn't live in the main repository yet does not mean it is not valuable.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
- Extensions are discoverable in a repeatable manner by all users of the primary project; just because code doesn't live in the main repository yet does not mean it is not valuable.
```

**Review by robtuley — APPROVED** [review]
I think this looks like a good forward iteration on this pattern. 

I can help add to the known instances here too as we have a internal inner source product here at Flutter that is undergoing exactly this pattern: a high growth area of the group has thrown 3 new dedicated teams at it to drive features and momentum, this caused all sorts of bottlenecks in existing contribution flow, and now the teams are re-achitecting to build themselves a plugin mechanism so they can experiments with features in a more independent and sandboxed way.

**Comment by Ssukriti** [comment]
@robtuley thank you. Yes that would be great to have more known instances :)

File: assets/img/extensions-for-sustainable-growth/README.md [hunk 12]
```diff
@@ -0,0 +1 @@
+If you want to edit the illustration, please request access to this [source document](https://docs.google.com/presentation/d/1X5c2dRFfJwQ7iBKbDKUM4jq_G0qm7uEK/edit#slide=id.p1).
```
Comment (spier) [inline assets/img/extensions-for-sustainable-growth/README.md]:
```suggestion
If you want to edit the illustration, please request access to this [source document](https://docs.google.com/presentation/d/1uyvRcIIrxKedajB6XeGL21_2p60oGOi-nrU6pwk5YqA/edit).
```

Ported this to a new doc.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 13]
```diff
@@ -1,23 +1,39 @@
...
-- Adding an excessive number of capabilities and code to the strategic repository is making it difficult to maintain.
-- As the maintainers cannot keep up with feedback to the contributors and code reviews anymore it creates a growing backlog of new features and ideas for the project which has scaled.
+- A strategic InnerSource codebase is scaling rapidly with new feature contributions from several employees.
+- The ratio of reviewers to contributions results in a growing backlog of pull requests. This is slowing down release of new features to the community.
+- Quality of the codebase is no longer maintained and user experience is adversely impacted.
+- Maintainers of the codebase are burdened and cannot keep up with the influx of contributions and increased community support.
+- There are more features being contributed than gaining adoption by users. Not all ideas are serving internal use cases, but addition of features is adding to the maintenance overhead.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
- Some of the contributed features are not gaining adoption by users, and might even turn fully dormant. However even though they are unused, these features are still adding to the maintenance overhead.
```

Would this be a simpler way to say the same thing?

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 14]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
    - Create criteria for when an extension can be ported to the primary repository, such as adoption of the extension by internal products and demand for the feature.
    - Porting process of the extension to the primary repository will follow more stringent code review guidelines as set by library maintainers.
 
+![Project extensions](../../assets/img/extensions-for-sustainable-growth/extensions-for-sustainable-growth.png)
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]: In the published pattern the text "Project extensions" will be displayed as a caption below the image.
So please put here whatever you want to be shown for that purpose.

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 15]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
...
+
 Following the principles outlined above ensures that:
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion

Following these principles ensures that:
```

simplifying

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 16]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
...
 
-- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of boilerplate code.
+- The addition of the extensions pattern with easy to use templates enables developers to add new features to a project's ecosystem without requiring them to write large amounts of [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) code.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]:
```suggestion
- Developers can add new features to a project's ecosystem without requiring them to write large amounts of [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) code.
```

I would not talk about the "pattern" at this point, as that is not how they users would think about it.
They might think "I added an extension using a template" but probably not "I used the extension pattern" :)

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 17]
```diff
@@ -54,17 +70,19 @@ In order for the extensions model to be successful, there are few architectural
...
 - Extensions are discoverable in a repeatable manner to all users of the primary project; just because code doesn't live in the main repository yet does not mean it is not valuable.
 - The maintainer burden is reduced until an extension has demonstrated that it fills an important gap in the primary project.
 - The core project's common code (e.g. base classes and utility functions) can be a starting point for new development that extends project's domain. This avoids the need to port innovative work after-the-fact, thus reducing the overall burden of developing novel features for the project.
 - Developers are more likely to contribute and stay involved in maintenance and building communities for their codebase, which is also good for the health of the overall project ecosystem.
 
 ## Resulting Context
 
-- The project is able to scale with the addition of new features, without adding a maintenance overhead on the primary project repository.
+- The project is able to scale with the addition of new features, without adding maintenance overhead on the primary project repository.
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md:89]: Is there really zero maintenance work created when using the extensions mechanism?
Or is it rather that the maintenance that is required for this to work seems more fruitful and scales better?

File: patterns/1-initial/extensions-for-sustainable-growth.md [hunk 18]
```diff
@@ -1,23 +1,39 @@
...
+As the number of contributions to a mature InnerSource repository rapidly increases, it adds more burden on code reviews and maintenance. This results in a large code review backlog or premature rejection of new feature contributions. How do we allow for faster release of new features, encouraging innovation and experimentation; while also keeping the repository well maintained?
+
+## Story
+
+There is a strategic project that aims to collect the best innovations within a domain space to one common stack, allowing reuse of a common infrastructure and providing a standard user experience. Through InnerSource, various teams in the organization that work within the domain space get an opportunity to collaborate and contribute their innovations to the common codebase. However, a large number of contributions in parallel from several developers is making maintenance of the codebase difficult. This is adding a huge burden on the project maintainers who assume ownership over the code quality standards and enable the community through various forms of communication.
+
+Project maintainers are burnt out due to:
```
Comment (spier) [inline patterns/1-initial/extensions-for-sustainable-growth.md]: Do you literally mean "burnout" here, or could we use a different phrase?

Just wondering if burnout might be too strong? 
However if that is how it was for the group in your story, by all means, keep it as is.

**Review by spier — COMMENTED** [review]
Great work @Ssukriti!
Please see my comments in line.

Once you are done reviewing the feedback, you can already move this pattern to the `2-structured` folder.

When you do that, you can also add your pattern to the mindmap already, by following [these instructions](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/pattern-categorization/README.md#adding-a-pattern-to-the-mind-map).

My only suggestion on the info in the mind map is to add something about the maintenance overhead e.g. 
> + scale -> (new category) Technical Challenges -> Exploding maintenance overhead -> our pattern

Once you are done, please let me know.

I will take another pass at the pattern and see if I find anything surprizing when reading it in one go.

Then we work in the final version of the visual (that we are working on in parallel), and then this should be good to go live 🚀

**Comment by spier** [comment]
@Ssukriti and one more thing.

We also discussed possible similarities between your pattern in this PR and the [Incubator Pipeline](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/incubator-pipeline.md).

My take is: 
Let's first get your pattern published in our book.
If we then get to publish the Incubator Pipeline to the book at a later point, we have to work out possible duplications then.

So in short: Just don't worry about this :)

**Comment by spier** [comment]
@robtuley thanks for chiming in !
What do you think about adding your Known Instance in a PR straight after this pattern goes live in the book? i.e. directly after this PR here is finished? 

Could make for a nice follow-up announcement, showcasing how the known instance information can provide value, and motivating other orgs to share their stories too.

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 9 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md:53]: hmm so not all new capabilities will move into the primary repository and project. Only ones that meet users needs. So that way extensions are filtering out what new capabilities make it into core of project.

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 17 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md:89]: since extensions lie outside project, they would not add a maintenance overhead to primary project repository . They will be independent repositories with the new features

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 6 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md]: I will remove experimental features here, since all new features are meant for exploration/ experimentation

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 18 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md]: I think burnout would be an appropriate term here that would be caused because of reasons mentioned. I could also remove this paragraph , since the matter is described above `However, a large number of contributions in parallel from several developers is making maintenance of the codebase difficult. This is adding a huge burden on the project maintainers who assume ownership over the code quality standards and enable the community through various forms of communication.`

and essentially we would like extensions pattern to be applied before maintainers face burnout , when maintenance starts to become difficult for a project

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 10 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md:104]: @spier just few unresolved comments left, let me know what you think and if answers are clear. I will then merge the PR and move to structured. Thank you

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 18 above)
Comment (spier) [reply patterns/1-initial/extensions-for-sustainable-growth.md]: Feel free to leave the burnout term in there, if this is the extent that you observed in your story.

We could also change it to this. Might make it a bit less strong?
> Project maintainers are at risk of burnout due to:

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 9 above)
Comment (spier) [reply patterns/1-initial/extensions-for-sustainable-growth.md:53]: Got it. Let's leave it as is.

**Review by spier — APPROVED** [review]
@Ssukriti all the text looks good to me now. Fantastic work!

Only questions remaining from my end:

1. Did you take a look at the modified visual? If you like it, would you mind integrating it into this PR?
2. Did I understand you correctly in that you want to first merge this PR, and then send another PR to move the file and add the pattern to the mindmap?

**Comment by Ssukriti** [comment]
@spier thank you for the edits to the diagram, I have updated the readme and image.
I do not know why the link check is failing, the links are correct.
2. Yes, could you please merge this PR and then I will open new one to move it to structured and add to mindmap

File: patterns/1-initial/extensions-for-sustainable-growth.md (hunk 18 above)
Comment (Ssukriti) [reply patterns/1-initial/extensions-for-sustainable-growth.md]: Good idea, updated

**Comment by spier** [comment]
@Ssukriti not sure what happened to the link check either.

PR is merged 🚀
